lesson1-oop-1/Pizza.ts

```diff
@@ -0,0 +1,19 @@
+import Consumable from './Consumable';
+
+export default class Pizza extends Consumable {
+  slicesEaten: number;
+
+  constructor(protected numberOfSlices: number, protected spoiled: boolean) {
+    super('pizza', 50, 12, spoiled);
+  }
+
+  eat(): string {
+    if(this.slicesEaten < this.numberOfSlices) {
+      this.slicesEaten++;
+      this.consumed = true;
+    }
+    return this.slicesEaten < this.numberOfSlices
+      ? `You eat the ${ this.name }.${ this.spoiled ? 'You feel sick.' : '' }`
```

Comment:
This string "`You eat the ${ this.name }." should be extracted from parent eta call.